music_assistant/server/providers/plex/__init__.py

```diff
@@ -0,0 +1,467 @@
+        plex_album = await self._get_data(prov_album_id, PlexAlbum)
+
+        tracks = []
+        for plex_track in await self._run_async(plex_album.tracks):
+            track = await self._parse_track(plex_track)
+            tracks.append(track)
+        return tracks
+
+    async def get_artist(self, prov_artist_id) -> Artist:
+        """Get full artist details by id."""
+        plex_artist = await self._get_data(prov_artist_id, PlexArtist)
+        return await self._parse_artist(plex_artist) if plex_artist else None
+
+    async def get_track(self, prov_track_id) -> Track:
+        """Get full track details by id."""
+        plex_track = await self._get_data(prov_track_id, PlexTrack)
+        return await self._parse_track(plex_track)
+
+    async def get_playlist(self, prov_playlist_id) -> Playlist:
+        """Get full playlist details by id."""
+        plex_playlist = await self._get_data(prov_playlist_id, PlexPlaylist)
+        return await self._parse_playlist(plex_playlist)
+
+    async def get_playlist_tracks(  # type: ignore[return]
+        self, prov_playlist_id: str
+    ) -> AsyncGenerator[Track, None]:
+        """Get all playlist tracks for given playlist id."""
+        plex_playlist = await self._get_data(prov_playlist_id, PlexPlaylist)
+
+        playlist_items = await self._run_async(plex_playlist.items)
+
+        if not playlist_items:
+            yield None
+        for index, plex_track in enumerate(playlist_items):
+            track = await self._parse_track(plex_track)
+            if track:
+                track.position = index + 1
+                yield track
+
+    async def get_artist_albums(self, prov_artist_id) -> list[Album]:
+        """Get a list of albums for the given artist."""
+        plex_artist = await self._get_data(prov_artist_id, PlexArtist)
+        plex_albums = await self._run_async(plex_artist.albums)
+        if plex_albums:
+            albums = []
+            for album_obj in plex_albums:
+                albums.append(await self._parse_album(album_obj))
+            return albums
+        return []
+
+    async def get_stream_details(self, item_id: str) -> StreamDetails | None:
+        """Get streamdetails for a track."""
+        plex_track = await self._get_data(item_id, PlexTrack)
+        if not plex_track or not plex_track.media:
+            raise MediaNotFoundError(f"track {item_id} not found")
+
+        media: PlexMedia = plex_track.media[0]
+        media_part: PlexMediaPart = media.parts[0]
+        audio_stream: PlexAudioStream = media_part.audioStreams()[0]
+
+        media_type = ContentType.try_parse(media.container)
+
+        stream_details = StreamDetails(
+            item_id=plex_track.key,
+            provider=self.domain,
+            content_type=ContentType.try_parse(media.container),
+            duration=plex_track.duration,
+            channels=media.audioChannels,
+            data=plex_track,
+            loudness=audio_stream.loudness,
```

Comment:
scrap if this is not an EBU-R128 measurement, normally this is measured in the stream controller.
Only when you are certain you can return the EBU-R128 average loudness as float, you may set it here yourself, otherwise leave it out and it will be auto filled.

Reply:
It is, but I just figured out that the analysis seems to done for plex pass owners only, so I will remove it

Reply:
ah thats a bummer. But if it is EBU-R128 we could set it if exists ?

Reply:
At least I assume it has a value when the user has a plex pass and no value if otherwise

Reply:
Okay, I added it if present 👍